cache/clock_cache.cc

```diff
@@ -117,19 +117,80 @@ void ClockHandleTable::Assign(ClockHandle* dst, ClockHandle* src) {
   dst->key_data = src->key_data;
   dst->flags.store(0);
   dst->SetIsElement(true);
-  dst->SetClockPriority(ClockHandle::ClockPriority::NONE);
   dst->SetCachePriority(src->GetCachePriority());
+  // dst->SetClockPriority(ClockHandle::ClockPriority::NONE);
   occupancy_++;
 }
 
-int ClockHandleTable::FindElement(const Slice& key, uint32_t hash,
-                                  uint32_t& probe) {
-  return FindSlot(
+bool ClockHandleTable::TryRemove(ClockHandle* h,
+                                 autovector<ClockHandle>* deleted) {
+  if (h->TryExclusiveRef()) {
+    if (h->WillBeDeleted()) {
+      Remove(h, deleted);
+      return true;
+    }
+    h->ReleaseExclusiveRef();
+  }
+  return false;
+}
+
+bool ClockHandleTable::SpinTryRemove(ClockHandle* h,
+                                     autovector<ClockHandle>* deleted) {
+  if (h->SpinTryExclusiveRef()) {
+    if (h->WillBeDeleted()) {
+      Remove(h, deleted);
+      return true;
+    }
+    h->ReleaseExclusiveRef();
+  }
+  return false;
+}
+
+void ClockHandleTable::ClockOff(ClockHandle* h) {
+  h->SetClockPriority(ClockHandle::ClockPriority::NONE);
+}
+
+void ClockHandleTable::ClockOn(ClockHandle* h) {
+  assert(!h->IsInClock());
+  bool is_high_priority =
+      h->HasHit() || h->GetCachePriority() == Cache::Priority::HIGH;
+  h->SetClockPriority(static_cast<ClockHandle::ClockPriority>(
+      is_high_priority * ClockHandle::ClockPriority::HIGH +
```

Comment:
Why not ```is_high_priority ? ClockHandle::ClockPriority::HIGH : ClockHandle::ClockPriority::MEDIUM```? Might be a little easier to reason about.

Reply:
I did it this way to avoid a conditional. But I guess I should trust the compiler more?